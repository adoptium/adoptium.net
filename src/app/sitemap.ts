import type { MetadataRoute } from "next";
import { getAppRoutes, getBlogRoutes } from "@/utils/getAppRoutes";
import { routing } from "@/i18n/routing";
import { getAllStaticSitemapUrls } from "@/utils/sitemapData";
import fs from "fs";
import path from "path";
import authorsData from "@/data/authors.json";

export default function sitemap(): MetadataRoute.Sitemap {
  const baseUrl = process.env.NEXT_PUBLIC_SITE_URL || "https://adoptium.net";

  const staticRoutes = getAppRoutes().filter(
    (route) => !route.includes("[...slug]") && !route.startsWith("/news")
  );

  // Getting additional routes from sitemap data to ensure all proposal URLs are included
  const sitemapUrls = getAllStaticSitemapUrls();
  const additionalRoutes = sitemapUrls
    .map(url => url.replace(/\/$/, '')) // Removing trailing slashes for consistency
    .filter(url => !staticRoutes.includes(url)); // Adding routes not already added in static routes

  const allStaticRoutes = [...staticRoutes, ...additionalRoutes];

  const blogRoutes = getBlogRoutes();

  // Collect asciidoc routes from content/asciidoc-pages (recursive)
  function getAsciidocRoutes(dir: string, prefix = ""): string[] {
    let routes: string[] = [];
    const entries = fs.readdirSync(dir, { withFileTypes: true });
    for (const entry of entries) {
      if (entry.isDirectory()) {
        routes = routes.concat(
          getAsciidocRoutes(
            path.join(dir, entry.name),
            `${prefix}/${entry.name}`
          )
        );
      } else if (entry.isFile() && entry.name.endsWith(".adoc")) {
        const route = `${prefix}/${entry.name.replace(/\index.adoc$/, "")}`;
        routes.push(route);
      }
    }
    return routes;
  }
  const asciidocDir = path.join(process.cwd(), "content/asciidoc-pages");
  let asciidocRoutes: string[] = [];
  try {
    asciidocRoutes = getAsciidocRoutes(asciidocDir);
  } catch {}

  const entries: MetadataRoute.Sitemap = [];

  // Build a set of canonical URLs handled by asciidoc index pages
  const asciidocCanonicalSet = new Set<string>();
  for (const route of asciidocRoutes) {
    const match = route.match(/(.+?)\/index(?:\.(\w{2,5}))?\.adoc$/);
    if (match) {
      const canonical = `${match[1]}/`;
      asciidocCanonicalSet.add(canonical);
    }
  }

  // Static pages (localized), but skip those handled by asciidoc index pages
  for (const route of allStaticRoutes) {
    const isRoot = route === "/";
    const canonical = isRoot ? "/" : route.endsWith("/") ? route : `${route}/`;
    if (!isRoot && asciidocCanonicalSet.has(canonical)) continue;
    const defaultUrl = isRoot ? `${baseUrl}/` : `${baseUrl}${route}`;
    const languages: Record<string, string> = {};
    for (const locale of routing.locales) {
      if (locale === routing.defaultLocale) continue;
      languages[locale] = isRoot
        ? `${baseUrl}/${locale}`
        : `${baseUrl}/${locale}${route}`;
    }
    // Add x-default hreflang pointing to the default locale
    if (Object.keys(languages).length > 0) {
      languages["x-default"] = defaultUrl;
    }
    entries.push({
      url: defaultUrl,
      lastModified: undefined,
      alternates: Object.keys(languages).length > 0 ? { languages } : undefined,
    });
  }

  // Blog/news pages (not localized, with date)
  for (const blog of blogRoutes) {
    const url = `${baseUrl}${blog.loc}`;
    let lastModified: Date | undefined = undefined;
    const match = blog.loc.match(/\/news\/(\d{4})\/(\d{2})/);
    if (match) {
      const [year, month] = match.slice(1);
      lastModified = new Date(Number(year), Number(month) - 1, 1);
    }
    entries.push({
      url,
      lastModified,
    });
  }

  // Author pages (localized)
  const authorRoutes = getAuthorRoutes();
  for (const route of authorRoutes) {
    const defaultUrl = `${baseUrl}${route}`;
    const languages: Record<string, string> = {};
    for (const locale of routing.locales) {
      if (locale === routing.defaultLocale) continue;
      languages[locale] = `${baseUrl}/${locale}${route}`;
    }
    // Add x-default hreflang pointing to the default locale
    if (Object.keys(languages).length > 0) {
      languages["x-default"] = defaultUrl;
    }
    entries.push({
      url: defaultUrl,
      lastModified: undefined,
      alternates: Object.keys(languages).length > 0 ? { languages } : undefined,
    });
  }

  // Asciidoc pages (localized by file extension)
  // Group by canonical path (parent dir) and collect language variants
  const asciidocMap: Record<string, Record<string, string>> = {};
  for (const route of asciidocRoutes) {
    // Match /foo/bar/index(.xx)?.adoc
    const indexMatch = route.match(/(.+?)\/index(?:\.(\w{2,5}))?\.adoc$/);
    if (indexMatch) {
      const canonical = `${indexMatch[1]}/`;
      const lang = indexMatch[2] || routing.defaultLocale;
      if (!asciidocMap[canonical]) asciidocMap[canonical] = {};
      asciidocMap[canonical][lang] = `${canonical}`;
      continue;
    }
    // Otherwise, treat as a regular page (no index)
    const match = route.match(/(.+?)(?:\.(\w{2,5}))?$/);
    if (!match) continue;
    const base = match[1];
    const lang = match[2] || routing.defaultLocale;
    if (!asciidocMap[base]) asciidocMap[base] = {};
    asciidocMap[base][lang] = `${base}${match[2] ? "." + match[2] : ""}`;
  }
  for (const base in asciidocMap) {
    const langs = asciidocMap[base];
    if (!langs[routing.defaultLocale]) continue;
    // Canonical URL: always ends with a slash
    let normalizedBase = base.startsWith("/") ? base : `/${base}`;
    normalizedBase = normalizedBase.replace(/\/+/g, "/");
    const defaultUrl = `${baseUrl}${normalizedBase}`;
    const languages: Record<string, string> = {};
    for (const locale of routing.locales) {
      if (locale === routing.defaultLocale) continue;
      if (langs[locale]) {
        languages[locale] = `${baseUrl}/${locale}${normalizedBase}`;
      }
    }
    // Add x-default hreflang pointing to the default locale
    if (Object.keys(languages).length > 0) {
      languages["x-default"] = defaultUrl;
    }
    entries.push({
      url: defaultUrl,
      lastModified: undefined,
      alternates: Object.keys(languages).length > 0 ? { languages } : undefined,
    });
  }

  // Collect author routes from authors.json
  function getAuthorRoutes(): string[] {
    try {
      return Object.keys(authorsData).map(
        (authorId) => `/news/author/${authorId}/`
      );
    } catch {
      return [];
    }
  }

  return entries;
}
