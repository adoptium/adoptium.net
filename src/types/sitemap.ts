export interface SitemapPage {
  title: string;
  url: string;
}

export interface SitemapSection {
  id: string;
  title: string;
  description?: string;
  pages: SitemapPage[];
}

export interface SitemapData {
  sections: SitemapSection[];
  dynamicContent?: {
    recentBlogs?: SitemapPage[];
  };
}

export const Sections = {
  topLevel: 'top-level',
  temurin: 'temurin',
  installation: 'installation',
  marketplace: 'marketplace',
  documentation: 'documentation',
  aqavit: 'aqavit',
  security: 'security',
  community: 'community',
  brand: 'brand',
  support: 'support',
} as const;

export type SectionId = typeof Sections[keyof typeof Sections];