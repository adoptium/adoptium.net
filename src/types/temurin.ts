// Common types for Temurin releases across the application
export interface VersionMetaData {
  major: number;
  minor: number;
  security: number;
  pre?: string;
  patch?: number;
  build: number;
  optional?: string;
  adopt_build_number?: number;
  openjdk_version: string;
  semver: string;
}

export interface Binary {
  type: string;
  link: string;
  checksum: string;
  size: number;
  extension: string;
  release_name: string;
  installer_link?: string;
  installer_checksum?: string;
  installer_size?: number;
  installer_extension?: string;
}

export interface ReleaseAsset {
  platform_name: string;
  os: string;
  architecture: string;
  release_name: string;
  release_link: string;
  release_date: string;
  version: VersionMetaData;
  binaries: Array<Binary>;
}
