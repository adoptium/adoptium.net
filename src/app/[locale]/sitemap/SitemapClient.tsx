'use client';

import { useState, useMemo, useEffect } from 'react';
import { Link } from '@/i18n/navigation';
import { SitemapData, SitemapPage } from '@/types/sitemap';
import {
  FaHome,
  FaDownload,
  FaCog,
  FaStore,
  FaBook,
  FaCheckCircle,
  FaShieldAlt,
  FaUsers,
  FaPalette,
  FaHeart,
  FaNewspaper,
  FaChevronDown,
  FaChevronRight,
  FaSearch
} from 'react-icons/fa';
import { Sections } from '@/types/sitemap';

function useDebounce<T>(value: T, delay: number): T {
  const [debouncedValue, setDebouncedValue] = useState<T>(value);

  useEffect(() => {
    const handler = setTimeout(() => {
      setDebouncedValue(value);
    }, delay);

    return () => {
      clearTimeout(handler);
    };
  }, [value, delay]);

  return debouncedValue;
}

interface SectionWrapperProps {
  icon: React.ComponentType<{ className?: string }>;
  title: string;
  description?: string;
  children: React.ReactNode;
  isExpanded?: boolean;
  onToggle: () => void;
}

function SectionWrapper({ 
  icon: IconComponent, 
  title, 
  description, 
  children, 
  isExpanded = false, 
  onToggle 
}: SectionWrapperProps) {
  const headerContent = (
    <div className="flex items-center gap-4">
      <IconComponent className="text-2xl text-rose-500" />
      <div>
        <h2 className="text-xl font-semibold text-white">
          {title}
        </h2>
        {description && (
          <p className="text-sm text-gray-400 mt-1">
            {description}
          </p>
        )}
      </div>
    </div>
  );

  return (
    <div className="bg-gray-800/50 backdrop-blur-sm rounded-lg border border-gray-700 overflow-hidden">
        <button
          onClick={onToggle}
          className="w-full px-6 py-4 text-left hover:bg-gray-700/50 transition-colors duration-200 flex items-center justify-between"
        >
          {headerContent}
          {isExpanded ? (
            <FaChevronDown className="text-gray-400" />
          ) : (
            <FaChevronRight className="text-gray-400" />
          )}
        </button>

      {isExpanded && (
        <div className="px-6 pb-4 pt-2">
          {children}
        </div>
      )}
    </div>
  );
}

interface SitemapClientProps {
  sitemapData: SitemapData;
}

const sectionIcons: Record<string, React.ComponentType<{ className?: string }>> = {
  [Sections.topLevel]: FaHome,
  [Sections.temurin]: FaDownload,
  [Sections.installation]: FaCog,
  [Sections.marketplace]: FaStore,
  [Sections.documentation]: FaBook,
  [Sections.aqavit]: FaCheckCircle,
  [Sections.security]: FaShieldAlt,
  [Sections.community]: FaUsers,
  [Sections.brand]: FaPalette,
  [Sections.support]: FaHeart,
};

export default function SitemapClient({ sitemapData }: SitemapClientProps) {
  const [searchTerm, setSearchTerm] = useState('');
  const [expandedSections, setExpandedSections] = useState<Set<string>>(new Set());
  const [expandedDynamicSections, setExpandedDynamicSections] = useState<Set<string>>(new Set());

  const debouncedSearchTerm = useDebounce(searchTerm, 300);


  const filteredSections = useMemo(() => {
    if (!debouncedSearchTerm.trim()) return sitemapData.sections;

    const searchLower = debouncedSearchTerm.toLowerCase();
    return sitemapData.sections.map(section => ({
      ...section,
      pages: section.pages.filter(page =>
        page.title.toLowerCase().includes(searchLower)
      )
    })).filter(section => section.pages.length > 0);
  }, [sitemapData.sections, debouncedSearchTerm]);

  const filteredDynamicContent = useMemo(() => {
    if (!debouncedSearchTerm.trim() || !sitemapData.dynamicContent) {
      return sitemapData.dynamicContent;
    }

    const searchLower = debouncedSearchTerm.toLowerCase();
    return {
      recentBlogs: sitemapData.dynamicContent.recentBlogs?.filter(blog =>
        blog.title.toLowerCase().includes(searchLower)
      ),
    };
  }, [sitemapData.dynamicContent, debouncedSearchTerm]);

  const toggleSection = (sectionId: string) => {
    setExpandedSections(prev => {
      const newSet = new Set(prev);
      if (newSet.has(sectionId)) {
        newSet.delete(sectionId);
      } else {
        newSet.add(sectionId);
      }
      return newSet;
    });
  };

  const toggleDynamicSection = (sectionId: string) => {
    setExpandedDynamicSections(prev => {
      const newSet = new Set(prev);
      if (newSet.has(sectionId)) {
        newSet.delete(sectionId);
      } else {
        newSet.add(sectionId);
      }
      return newSet;
    });
  };

  const renderPageLink = (page: SitemapPage) => {
    const isExternal = /^https?:\/\//.test(page.url);
    const linkClasses = `block py-2 px-4 text-white hover:text-rose-600 transition-colors duration-200`;

    if (isExternal) {
      return (
        <a
          href={page.url}
          target="_blank"
          rel="noopener noreferrer"
          className={linkClasses}
        >
          {page.title}
        </a>
      );
    }

    return (
      <Link href={page.url} className={linkClasses}>
        {page.title}
      </Link>
    );
  };

  return (
    <div className="min-h-screen bg-purple">
      <div className="container mx-auto px-4 py-8">
        {/* Search Bar */}
        <div className="max-w-2xl mx-auto mb-8">
          <div className="relative">
            <FaSearch className="absolute left-4 top-1/2 transform -translate-y-1/2 text-gray-400" />
            <input
              type="text"
              placeholder="Search pages..."
              value={searchTerm}
              onChange={(e) => setSearchTerm(e.target.value)}
              className="w-full pl-12 pr-4 py-3 bg-gray-800 border border-gray-600 rounded-lg text-white placeholder-gray-400 focus:outline-none focus:ring-2 focus:ring-rose-500 focus:border-transparent"
            />
          </div>
        </div>

        {/* Sitemap Sections */}
        <div className="max-w-6xl mx-auto space-y-6">
          {filteredSections.map((section) => {
            const IconComponent = sectionIcons[section.id] || FaHome;
            const isExpanded = expandedSections.has(section.id);

            return (
              <SectionWrapper
                key={section.id}
                icon={IconComponent}
                title={section.title}
                description={section.description}
                isExpanded={isExpanded}
                onToggle={() => toggleSection(section.id)}
              >
                <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-2">
                  {section.pages.map((page) => (
                    <div key={page.url} className="group">
                      {renderPageLink(page)}
                    </div>
                  ))}
                </div>
              </SectionWrapper>
            );
          })}

          {/* Dynamic Content Sections */}
          {filteredDynamicContent && (
            <>
              {/* Recent Blogs */}
              {filteredDynamicContent.recentBlogs && filteredDynamicContent.recentBlogs.length > 0 && (
                <SectionWrapper
                  icon={FaNewspaper}
                  title="Latest News"
                  description="Recent blog posts and announcements"
                  isExpanded={expandedDynamicSections.has('recentBlogs')}
                  onToggle={() => toggleDynamicSection('recentBlogs')}
                >
                  <div className="grid grid-cols-1 md:grid-cols-2 gap-2">
                    {filteredDynamicContent.recentBlogs.map((blog) => (
                      <div key={blog.url} className="group">
                        {renderPageLink(blog)}
                      </div>
                    ))}
                  </div>
                </SectionWrapper>
              )}
            </>
          )}
        </div>
      </div>
    </div>
  );
}
