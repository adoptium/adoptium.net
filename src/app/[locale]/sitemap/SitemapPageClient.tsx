'use client';

import { useTranslations } from "next-intl";
import PageHeader from "@/components/Common/PageHeader";
import SitemapClient from "./SitemapClient";
import { SitemapData } from "@/types/sitemap";

interface SitemapPageClientProps {
  sitemapData: SitemapData;
}

export default function SitemapPageClient({ sitemapData }: SitemapPageClientProps) {
  const t = useTranslations("Sitemap");

  return (
    <div>
      <PageHeader
        title={t("title")}
        subtitle={t("subtitle")}
        description={t("description")}
      />
      <SitemapClient sitemapData={sitemapData} />
    </div>
  );
}
