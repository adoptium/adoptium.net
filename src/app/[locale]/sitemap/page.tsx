import { Metadata } from 'next';
import { generateSitemapData } from '@/utils/sitemapData';
import SitemapPageClient from './SitemapPageClient';

export const metadata: Metadata = {
  title: 'Sitemap',
  description: 'Complete sitemap of the Adoptium website including all pages, documentation, and resources.',
  robots: {
    index: true,
    follow: true,
  },
};

export default async function SitemapPage() {
  const sitemapData = generateSitemapData();

  return <SitemapPageClient sitemapData={sitemapData} />;
}
