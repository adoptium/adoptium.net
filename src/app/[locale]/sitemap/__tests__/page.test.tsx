import { render, screen, fireEvent } from '@testing-library/react';
import SitemapClient from '../SitemapClient';
import { describe, expect, it, vi, beforeEach, afterEach } from 'vitest';
import { SitemapData } from '@/types/sitemap';


describe('SitemapClient', () => {
  beforeEach(() => {
    vi.clearAllMocks();
    vi.useFakeTimers();
    document.body.innerHTML = '';
  });

  afterEach(() => {
    vi.useRealTimers();
    document.body.innerHTML = '';
  });

  const renderSitemapClient = (customData?: SitemapData) => {
    document.body.innerHTML = '';
    const defaultData: SitemapData = {
      sections: [
        {
          id: 'top-level',
          title: 'Top-level',
          description: 'Main pages and navigation',
          pages: [
            { title: 'Home', url: '/' },
            { title: 'About', url: '/about' },
          ],
        },
      ],
      dynamicContent: {
        recentBlogs: [
          { title: 'Java 21 Release', url: '/news/2024/01/java-21-release' },
        ],
      },
    };

    return render(<SitemapClient sitemapData={customData || defaultData} />);
  };

  it('renders search input correctly', () => {
    renderSitemapClient();

    const searchInput = screen.getByPlaceholderText('Search pages...');
    expect(searchInput).toBeInTheDocument();
    expect(searchInput).toHaveValue('');
  });

  it('renders all sections and dynamic content sections', () => {
    renderSitemapClient();

    // Checking that section headers are rendered
    expect(screen.getByText('Top-level')).toBeInTheDocument();

    // Checking that dynamic content sections are rendered
    expect(screen.getByText('Latest News')).toBeInTheDocument();
  });

  it('has clickable section buttons', () => {
    renderSitemapClient();

    const topLevelButton = screen.getByRole('button', { name: /top-level/i });
    expect(topLevelButton).toBeInTheDocument();

    // Test passes if no errors are thrown when clicking
    fireEvent.click(topLevelButton);
  });

  it('renders sections with proper content', () => {
    renderSitemapClient();

    // Check that sections are rendered with proper content
    expect(screen.getByText('Top-level')).toBeInTheDocument();
    expect(screen.getByText('Latest News')).toBeInTheDocument();
  });

  it('handles empty dynamic content gracefully', () => {
    const dataWithoutDynamicContent = {
      sections: [
        {
          id: 'top-level',
          title: 'Top-level',
          description: 'Main pages and navigation',
          pages: [{ title: 'Home', url: '/' }],
        },
      ],
      dynamicContent: undefined,
    };

    renderSitemapClient(dataWithoutDynamicContent);

    expect(screen.queryByText('Latest News')).not.toBeInTheDocument();
  });

  it('handles empty sections gracefully', () => {
    const dataWithEmptySections = {
      sections: [],
      dynamicContent: {
        recentBlogs: [{ title: 'Java 21 Release', url: '/news/2024/01/java-21-release' }],
      },
    };

    renderSitemapClient(dataWithEmptySections);

    expect(screen.queryByText('Top-level')).not.toBeInTheDocument();
  });

  it('handles search input changes without crashing', async () => {
    renderSitemapClient();

    const searchInput = screen.getByPlaceholderText('Search pages...');

    // Checking various search inputs
    fireEvent.change(searchInput, { target: { value: 'test' } });
    fireEvent.change(searchInput, { target: { value: 'another test' } });
    fireEvent.change(searchInput, { target: { value: '' } });

    // Checking that the input should update
    expect(searchInput).toHaveValue('');
  });

  it('renders all expected sections with proper structure', () => {
    renderSitemapClient();

    // Checking that all expected elements are rendered
    expect(screen.getByPlaceholderText('Search pages...')).toBeInTheDocument();
    expect(screen.getByText('Top-level')).toBeInTheDocument();
    expect(screen.getByText('Latest News')).toBeInTheDocument();
    expect(screen.getByText('Top-level')).toBeInTheDocument();
  });

  it('renders basic component structure correctly', () => {
    const { container } = renderSitemapClient();
    expect(container.firstChild).toBeInTheDocument();
  });

  it('has no accessibility violations', async () => {
    const { container } = renderSitemapClient();

    const buttons = container.querySelectorAll('button');
    const inputs = container.querySelectorAll('input');
    const headings = container.querySelectorAll('h1, h2, h3, h4, h5, h6');

    buttons.forEach(button => {
      expect(button.textContent?.trim()).toBeTruthy();
    });

    inputs.forEach(input => {
      expect(input.getAttribute('placeholder')).toBeTruthy();
    });

    expect(headings.length).toBeGreaterThan(0);

    const interactiveElements = container.querySelectorAll('button, input, [tabindex]');
    expect(interactiveElements.length).toBeGreaterThan(0);
  });

  it('renders correctly', () => {
    const { container } = renderSitemapClient();
    expect(container).toMatchSnapshot();
  });
});
