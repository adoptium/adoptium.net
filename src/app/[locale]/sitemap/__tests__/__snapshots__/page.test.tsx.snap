// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SitemapClient > renders correctly 1`] = `
<div>
  <div
    class="min-h-screen bg-purple"
  >
    <div
      class="container mx-auto px-4 py-8"
    >
      <div
        class="max-w-2xl mx-auto mb-8"
      >
        <div
          class="relative"
        >
          <svg
            class="absolute left-4 top-1/2 transform -translate-y-1/2 text-gray-400"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            class="w-full pl-12 pr-4 py-3 bg-gray-800 border border-gray-600 rounded-lg text-white placeholder-gray-400 focus:outline-none focus:ring-2 focus:ring-rose-500 focus:border-transparent"
            placeholder="Search pages..."
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="max-w-6xl mx-auto space-y-6"
      >
        <div
          class="bg-gray-800/50 backdrop-blur-sm rounded-lg border border-gray-700 overflow-hidden"
        >
          <button
            class="w-full px-6 py-4 text-left hover:bg-gray-700/50 transition-colors duration-200 flex items-center justify-between"
          >
            <div
              class="flex items-center gap-4"
            >
              <svg
                class="text-2xl text-rose-500"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
                />
              </svg>
              <div>
                <h2
                  class="text-xl font-semibold text-white"
                >
                  Top-level
                </h2>
                <p
                  class="text-sm text-gray-400 mt-1"
                >
                  Main pages and navigation
                </p>
              </div>
            </div>
            <svg
              class="text-gray-400"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
              />
            </svg>
          </button>
        </div>
        <div
          class="bg-gray-800/50 backdrop-blur-sm rounded-lg border border-gray-700 overflow-hidden"
        >
          <button
            class="w-full px-6 py-4 text-left hover:bg-gray-700/50 transition-colors duration-200 flex items-center justify-between"
          >
            <div
              class="flex items-center gap-4"
            >
              <svg
                class="text-2xl text-rose-500"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M552 64H88c-13.255 0-24 10.745-24 24v8H24c-13.255 0-24 10.745-24 24v272c0 30.928 25.072 56 56 56h472c26.51 0 48-21.49 48-48V88c0-13.255-10.745-24-24-24zM56 400a8 8 0 0 1-8-8V144h16v248a8 8 0 0 1-8 8zm236-16H140c-6.627 0-12-5.373-12-12v-8c0-6.627 5.373-12 12-12h152c6.627 0 12 5.373 12 12v8c0 6.627-5.373 12-12 12zm208 0H348c-6.627 0-12-5.373-12-12v-8c0-6.627 5.373-12 12-12h152c6.627 0 12 5.373 12 12v8c0 6.627-5.373 12-12 12zm-208-96H140c-6.627 0-12-5.373-12-12v-8c0-6.627 5.373-12 12-12h152c6.627 0 12 5.373 12 12v8c0 6.627-5.373 12-12 12zm208 0H348c-6.627 0-12-5.373-12-12v-8c0-6.627 5.373-12 12-12h152c6.627 0 12 5.373 12 12v8c0 6.627-5.373 12-12 12zm0-96H140c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h360c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12z"
                />
              </svg>
              <div>
                <h2
                  class="text-xl font-semibold text-white"
                >
                  Latest News
                </h2>
                <p
                  class="text-sm text-gray-400 mt-1"
                >
                  Recent blog posts and announcements
                </p>
              </div>
            </div>
            <svg
              class="text-gray-400"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
